dask/typing.py

```diff
@@ -0,0 +1,364 @@
+from __future__ import annotations
+
+import abc
+from collections.abc import Callable, Hashable, Mapping, Sequence
+from typing import Any, Protocol, TypeVar, runtime_checkable
+
+try:
+    from IPython.display import DisplayObject
+except ImportError:
+    DisplayObject = Any
+
+
+CollType = TypeVar("CollType", bound="DaskCollection")
+CollType_co = TypeVar("CollType_co", bound="DaskCollection", covariant=True)
+PostComputeCallable = Callable
+
+
+class SchedulerGetCallable(Protocol):
+    """Protocol defining the signature of a __dask_scheduler__ callable."""
+
+    def __call__(
+        self,
+        dask: Mapping,
+        keys: Sequence[Hashable] | Hashable,
+        **kwargs: Any,
+    ) -> Any:
+        """Method called as the default scheduler for a collection.
+
+        Parameters
+        ----------
+        dsk : Mapping
+            The task graph.
+        keys : Sequence[Hashable] | Hashable
+            Key(s) corresponding to the desired data.
+        **kwargs : Any
+            Additional arguments.
+
+        Returns
+        -------
+        Any
+            Result(s) associated with `keys`
+
+        """
+        raise NotImplementedError("Inheriting class must implement this method.")
+
+
+class PostPersistCallable(Protocol[CollType_co]):
+    """Protocol defining the signature of a __dask_postpersist__ callable."""
+
+    def __call__(
+        self,
+        dsk: Mapping,
+        *args: Any,
+        rename: Mapping[str, str] | None = None,
```

Comment:
I believe these can be any keys, so `Mapping[Hashable, Hashable]` should be more correct, unless I'm missing something

Reply:
The way `__post_persist__`'s expected signature is described in the [custom collection docs](https://docs.dask.org/en/latest/custom-collections.html) it appears that this `Mapping` should be `str -> str`. If the keys `("a", 0), ("a", 1)` should be redefined to be `("b", 0), ("b", 1)`, use `rename={"a": "b"}`. (that is I don't think it's designed to redefine whole keys, just the string part)

Reply:
Ah, okay, my misunderstanding.